src/zenml/cli/stack_components.py

```diff
@@ -201,41 +228,165 @@ def register_stack_component_command(
+                f"Successfully registered {display_name} `{name}`."
+            )
 
     return register_stack_component_command
 
 
+def generate_stack_component_update_command(
+    component_type: StackComponentType,
+) -> Callable[[str, List[str]], None]:
+    """Generates an `update` command for the specific stack component type."""
+    display_name = _component_display_name(component_type)
+
+    @click.argument(
+        "name",
+        type=str,
+        required=True,
+    )
+    @click.argument("args", nargs=-1, type=click.UNPROCESSED)
+    def update_stack_component_command(name: str, args: List[str]) -> None:
+        """Updates a stack component."""
+        cli_utils.print_active_profile()
+        with console.status(f"Updating {display_name} '{name}'...\n"):
+            repo = Repository()
+            current_component = repo.get_stack_component(component_type, name)
+            if current_component is None:
+                cli_utils.error(f"No {display_name} found for name '{name}'.")
+
+            try:
+                parsed_args = cli_utils.parse_unknown_options(args)
+            except AssertionError as e:
+                cli_utils.error(str(e))
+                return
+            for prop in MANDATORY_COMPONENT_PROPERTIES:
+                if prop in parsed_args:
+                    cli_utils.error(
+                        f"Cannot update mandatory property '{prop}' of '{name}' {current_component.TYPE}. "
+                    )
+
+            from zenml.stack.stack_component_class_registry import (
+                StackComponentClassRegistry,
+            )
+
+            component_class = StackComponentClassRegistry.get_class(
+                component_type=component_type,
+                component_flavor=current_component.FLAVOR,
+            )
+            available_properties = _get_available_properties(component_class)
+            for prop in parsed_args.keys():
+                if (prop not in available_properties) and (
+                    len(available_properties) > 0
+                ):
+                    cli_utils.error(
+                        f"You cannot update the {display_name} `{current_component.name}` with property '{prop}'. You can only update the following properties: {available_properties}."
+                    )
+                elif prop not in available_properties:
+                    cli_utils.error(
+                        f"You cannot update the {display_name} `{current_component.name}` with property '{prop}' as this {display_name} has no optional properties that can be configured."
+                    )
+                else:
+                    continue
+
+            required_properties = _get_required_properties(component_class)
+            for prop in required_properties:
+                if prop not in parsed_args:
+                    parsed_args[prop] = getattr(current_component, prop)
+            updated_component = component_class(name=name, **parsed_args)
```

Comment:
If the `component.update` worked in the rename method, it might be worth trying it here as well like this:
```
updated_component = current_component.copy(update=parsed_args)
```